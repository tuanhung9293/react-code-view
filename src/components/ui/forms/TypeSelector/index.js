import React from 'react'
import Select from 'components/ui/forms/Select'
import {AVAILABLE_MEDIA_TYPES} from 'constants/defaultValues'

function TypeSelector(props) {
  return (
    <Select {...props}>
      {AVAILABLE_MEDIA_TYPES.map(type => (
        <option key={type.id} value={type.value}>{type.message}</option>
      ))}
    </Select>
  )
}

export default TypeSelector