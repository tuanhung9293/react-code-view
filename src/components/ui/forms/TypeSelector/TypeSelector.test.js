import React from 'react'
import {shallow} from 'enzyme'
import TypeSelector from './'


describe('renders', () => {
  it('renders TypeSelector', () => {
    expect(shallow(
        <TypeSelector />
    )).toMatchSnapshot()
  })

  it('TypeSelector should pass props down', () => {
    expect(shallow(
        <TypeSelector data1="data 1" data2="data 2" />
    )).toMatchSnapshot()
  })
})