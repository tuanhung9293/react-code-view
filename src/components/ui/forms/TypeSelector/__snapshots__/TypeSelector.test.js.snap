// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TypeSelector should pass props down 1`] = `
<Select
  data1="data 1"
  data2="data 2"
>
  <option
    key="image"
    value="image"
  >
    Image
  </option>
  <option
    key="video"
    value="video"
  >
    Video
  </option>
  <option
    key="audio"
    value="audio"
  >
    Audio
  </option>
</Select>
`;

exports[`renders renders TypeSelector 1`] = `
<Select>
  <option
    key="image"
    value="image"
  >
    Image
  </option>
  <option
    key="video"
    value="video"
  >
    Video
  </option>
  <option
    key="audio"
    value="audio"
  >
    Audio
  </option>
</Select>
`;
